mithril-core/src/error.rs

```diff
@@ -115,6 +115,14 @@ pub enum RegisterError {
     SerializationError,
 }
 
+/// Errors which can be outputted by an initializer.
+#[derive(Debug, Clone, thiserror::Error, PartialEq, Eq)]
+pub enum StmInitializerError {
+    /// NotRegistered error
+    #[error("Initializer not registered. Cannot participate as a signer.")]
+    NotRegistered,
+}
+
```

Comment:
We are still thinking how to best handle the errors in mithril-core, but we certainly have as a goal to minimise the number of errors. I think it would be better to include this type of error in the `RegisterError` type.